daft/dataframe/dataframe.py

```diff
@@ -1090,61 +1089,160 @@ def explode(self, *columns: ColumnInputType) -> "DataFrame":
         builder = self._builder.explode(parsed_exprs)
         return DataFrame(builder)
 
+    @DataframePublicAPI
+    def unpivot(
+        self,
+        ids: ManyColumnsInputType,
+        values: ManyColumnsInputType = [],
+        variable_name: str = "variable",
+        value_name: str = "value",
+    ) -> "DataFrame":
+        """Unpivots a DataFrame from wide to long format.
+
+        Example:
```

Comment:
Echoing [Jay](https://github.com/Eventual-Inc/Daft/pull/2183#discussion_r1584218592) here, can we get a more motivating example?